tsdb/index/tsi1/partition.go

```diff
@@ -1043,11 +1043,13 @@ func (p *Partition) compact() {
 			p.currentCompactionN++
 			go func() {
 				p.compactLogFile(logFile)
-				p.mu.Lock()
-				p.currentCompactionN--
-				p.levelCompacting[0] = false
-				p.mu.Unlock()
-				p.Compact()
+				defer func() {
+					p.mu.Lock()
+					p.currentCompactionN--
+					p.levelCompacting[0] = false
+					p.mu.Unlock()
+					p.Compact()
```

Comment:
The `p.Compact` call here leads to indirect tail-call recursion. Putting the call in a defer will add another level to the stack. Depending on how deep the recursion goes, adding this extra stack frame every time may cause problems. Do we think `p.compactLogFile` panics?

Reply:
More specifically, do we think `p.compactLogFile` panics, and then that is recovered higher up in the stack? Otherwise, I don't see this change making any difference.